### PR TITLE
Support GCS default service-account for PinotFS Plugin

### PR DESCRIPTION
**Pull request by phuchuynh9994**
## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [x] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [x] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

**Comment by fx19880617** [comment]
Thanks for submitting this PR! 
@elonazoulay Could you please also help review this!

File: pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java [hunk 1]
```diff
@@ -63,20 +63,19 @@
...
-    String gcpKey = config.getProperty(GCP_KEY);
     try {
-      storage = StorageOptions.newBuilder()
-          .setProjectId(projectId)
-          .setCredentials(GoogleCredentials.fromStream(Files.newInputStream(Paths.get(gcpKey))))
-          .build()
-          .getService();
+      StorageOptions.Builder storageBuilder = StorageOptions.newBuilder();
+      if (!isNullOrEmpty(config.getProperty(PROJECT_ID)) && !isNullOrEmpty(config.getProperty(GCP_KEY))) {
+        String projectId = config.getProperty(PROJECT_ID);
```
Comment (fx19880617) [inline pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:71]: Adding a log for the PROJECT_ID?

File: pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java [hunk 2]
```diff
@@ -63,20 +63,19 @@
...
+        String gcpKey = config.getProperty(GCP_KEY);
+        storageBuilder.setProjectId(projectId);
+        credentials = GoogleCredentials.fromStream(Files.newInputStream(Paths.get(gcpKey)));
+      } else {
+        credentials = GoogleCredentials.getApplicationDefault();
```
Comment (fx19880617) [inline pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:76]: Add a log to declare that it's using default credential

**Review by fx19880617 — APPROVED** [review]
overall lgtm
Thanks for adding this!

File: pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java [hunk 3]
```diff
@@ -63,20 +63,21 @@
...
-          .setProjectId(projectId)
-          .setCredentials(GoogleCredentials.fromStream(Files.newInputStream(Paths.get(gcpKey))))
-          .build()
-          .getService();
+      StorageOptions.Builder storageBuilder = StorageOptions.newBuilder();
+      if (!isNullOrEmpty(config.getProperty(PROJECT_ID)) && !isNullOrEmpty(config.getProperty(GCP_KEY))) {
+        LOGGER.info("Configs are: {}, {}", PROJECT_ID, config.getProperty(PROJECT_ID));
+        String projectId = config.getProperty(PROJECT_ID);
+        String gcpKey = config.getProperty(GCP_KEY);
+        storageBuilder.setProjectId(projectId);
```
Comment (fx19880617) [inline pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:74]: Just curious about this `storageBuilder.setProjectId(...)`

Shall we allow users to just specify `PROJECT_ID` and use default credential for `GCP_KEY`

Comment (fx19880617) [reply pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:74]: Also, shall we check the presence of `PROJECT_ID` always? Or it's part from the application default?

Comment (phuchuynh9994) [reply pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:74]: PROJECT_ID is a part of service account. 
I think in passing credentials manually way, we dont need PROOECT_ID also. 
Ref doc: https://cloud.google.com/docs/authentication/production#linux-or-macos

Comment (fx19880617) [reply pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java:74]: cool, thanks for the clarification!

**Pull request merged**